x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/esnative/ReservedRealm.java

```diff
@@ -212,19 +219,27 @@ public void users(ActionListener<Collection<User>> listener) {
     }
 
 
-    private void getUserInfo(final String username, ActionListener<ReservedUserInfo> listener) {
+    private void getUserInfo(final String username, @Nullable SecureString credentials, ActionListener<ReservedUserInfo> listener) {
         if (securityIndex.indexExists() == false) {
-            listener.onResponse(getDefaultUserInfo(username));
+            if (autoconfigured && username.equals(ElasticUser.NAME) && bootstrapUserInfo.verifyPassword(credentials)) {
+                nativeUsersStore.storeAutoconfiguredElasticUser(bootstrapUserInfo, listener);
+            } else {
+                listener.onResponse(getDefaultUserInfo(username));
+            }
         } else {
             nativeUsersStore.getReservedUserInfo(username, ActionListener.wrap((userInfo) -> {
                 if (userInfo == null) {
-                    listener.onResponse(getDefaultUserInfo(username));
+                    if (autoconfigured && username.equals(ElasticUser.NAME) && bootstrapUserInfo.verifyPassword(credentials)) {
```

Comment:
can we factor out 
```
if (autoconfigured && username.equals(ElasticUser.NAME) && bootstrapUserInfo.verifyPassword(credentials)) {
                        nativeUsersStore.storeAutoconfiguredElasticUser(bootstrapUserInfo, listener);
                    } else {
                        listener.onResponse(getDefaultUserInfo(username));
                    }
```

instead of copying it as I had suggested earlier on too ?